cloudwash/providers/azure.py

```diff
@@ -39,77 +39,114 @@ def _dry_vms(all_vms):
-                [dry_data["DISCS"]["delete"].append(ddisc) for ddisc in rdiscs]
-            return rdiscs
-
-        def dry_pips():
-            rpips = []
-            if settings.azure.criteria.public_ip.unassigned:
-                rpips = azure_client.list_free_pip()
-                [dry_data["PIPS"]["delete"].append(dpip) for dpip in rpips]
-            return rpips
-
-        def dry_resources(hours_old=None):
-            dry_data["RESOURCES"]["delete"] = azure_client.list_resources_from_hours_old(
-                hours_old=hours_old
-                or (settings.azure.criteria.resource_group.resources_sla_minutes / 60)
-            )
-            return dry_data["RESOURCES"]["delete"]
-
-        # Remove / Stop VMs
-        def remove_vms(avms):
-            # Remove VMs
-            [azure_client.get_vm(vm_name).delete() for vm_name in avms["delete"]]
-            # Stop VMs
-            [azure_client.get_vm(vm_name).stop() for vm_name in avms["stop"]]
-
-        # Actual Cleaning and dry execution
-        if kwargs["vms"] or kwargs["_all"]:
-            avms = dry_vms()
-            if not is_dry_run:
-                remove_vms(avms=avms)
-                logger.info(f"Stopped VMs: \n{avms['stop']}")
-                logger.info(f"Removed VMs: \n{avms['delete']}")
-                logger.info(f"Skipped VMs: \n{avms['skip']}")
-        if kwargs["nics"] or kwargs["_all"]:
-            rnics = dry_nics()
-            if not is_dry_run and rnics:
-                azure_client.remove_nics_by_search()
-                logger.info(f"Removed NICs: \n{rnics}")
-        if kwargs["discs"] or kwargs["_all"]:
-            rdiscs = dry_discs()
-            if not is_dry_run and rdiscs:
-                azure_client.remove_discs_by_search()
-                logger.info(f"Removed Discs: \n{rdiscs}")
-        if kwargs["pips"] or kwargs["_all"]:
-            rpips = dry_pips()
-            if not is_dry_run and rpips:
-                azure_client.remove_pips_by_search()
-                logger.info(f"Removed PIPs: \n{rpips}")
-        if kwargs["_all_rg"]:
-            rres = dry_resources()
-            if not is_dry_run and rres:
-                azure_client.remove_resource_group_of_old_resources(
-                    hours_old=(settings.azure.criteria.vm.sla_minutes / 60)
-                )
-                logger.info(f"Removed Resources: \n{rres}")
-        if is_dry_run:
-            echo_dry(dry_data)
+    data = ['VMS', 'NICS', 'DISCS', 'PIPS', 'RESOURCES', 'STACKS']
+    regions = settings.azure.auth.regions
+    groups = settings.azure.auth.resource_groups
+
+    if "all" in regions:
+        with compute_client("azure", azure_region="us-west", resource_group="foo") as azure_client:
+            regions = list(zip(*azure_client.list_region()))[0]
+
+    for region in regions:
+        if "all" in groups:
+            with compute_client("azure", azure_region=region, resource_group="foo") as azure_client:
```

Comment:
Is `resource_group="foo"` OK? Seems odd, but I don't know the logic of `compute_client`

Reply:
This parameter is only stored within wrapanapi for use with some operations like list/remove_resource within RG. Listing available groups/Regions should only need subscription_id and credentials.

Reply:
Thanks! Can you leave a brief comment about that in the code so that we can avoid confusion in the future.